### DISCUSSION TITLE
Error message: Command failed: unknown device dallassensor

### DISCUSSION THREAD

**Discussion by Hans-JR** [emsesp/EMS-ESP32#2033] [Q&A] [answered]
I moved from a normal ESP32 board to a LilyGo ESP32 board. Here I installed the latest Softare 3.7.0 dev 38. In Setting I selected the board Liligo S3 and "unchecked" the box "activate Analogsensor". On the hardwareside I connected a DS18B20 to pin GPIO18 with a 4,7k pullup. The sensor is being displayed under the tab "Sensors". However it constantly sends the message " [command] Command failed: unknown device dallassensor". Is there something wron in my settings?
![Error message](https://github.com/user-attachments/assets/c322137b-6ae3-455a-9c4b-c91b9e0ddee3)
![Temp-sensor](https://github.com/user-attachments/assets/d3cab764-9184-49a1-b537-9c3fbed8f837)

**Comment by proddy** [comment]
You probably went from 3.5.x to the latest dev skipping the 3.6.5 stable release? We renamed `dallassensor` in release 3.6.0 (https://github.com/emsesp/EMS-ESP32/blob/dev/CHANGELOG.md#important-breaking-changes-4). You may have some external scripts calling the EMS-ESP APIs with the old syntax.

**Reply by Hans-JR** [reply]
I am only using the ioBroker EMS-Adapter without MQTT (only via API) and NO other scripts.
I've contacted the ioBroker forum but haven't got any reply yet. The adapter version is 4.8.0 which is the latest one. If I disable the checkbox "Temperatursensor" in the adapter settings, the error is gone but the datapoint is not being updated anymore. I haven't tried the checkbox "use old dallas prefix" yet since I don't know what it stands for......

**Reply by Delta1977** [reply]
IIRC are the Dallas-Aensors 1-Wire bus sensors and not Analogsensors.

**Reply by Hans-JR** [reply]
Yes I know. I made a mistake in my first post. It should read "unchecked" the Analogsensors. In the meantime I've edited the post.

**Comment by MichaelDvP** [comment, marked as answer]
This is from the emsesp adapter. For compatibility with 3.6.0 it tries to fetch the api/dallassensor (line 391) and in second step fetch api/temperaturesensor (line 417). See here: https://github.com/tp1de/ioBroker.ems-esp/blob/8b7e55a6b336a3b48d436f4a0c9bb263847f327f/lib/ems.js#L391

The checkbox "use old prefix" only changes ioBroker entities and have no effect to reading from emsesp (line 437).
Maybe @tp1de can change the logic to avoid these errormessages in emsesp. E.g. try once and remember the working api-name for the temperaturesensors, than we have only one error message in the log.

**Reply by MichaelDvP** [reply]
Another possible solution: The ioBroker adapter reads `api/system/info` and uses the devcies array to read all `api/<device>/values`. 
That's a clever idea and ensures that new entities or renamings are always up to date.
Because this array only listes the ems-devices the adapter has options to read also temperaturesesensor (dallassensor), analogsensor, custom. The scheduler is actually not implemented. These device names are fixed in code and can not updated automatically. But if we add emsesp-devices to the device list there is no extra code/option needed to read these  devices and entities. 
I've tested this with [my test build](https://github.com/MichaelDvP/EMS-ESP32/releases/tag/test) and it works good (disabled all extra options in ioBroker).  
- Commit: https://github.com/MichaelDvP/EMS-ESP32/commit/f6b0f563c589363e6b013440308a6731e62dcbec
- My system/info devices section: [emsesp_system_devices.txt](https://github.com/user-attachments/files/17128840/emsesp_system_devices.txt)
- ioBroker options:
![grafik](https://github.com/user-attachments/assets/9435208b-45ca-45e6-8e87-fffdae282931)
- ioBroker shows up like this:
![grafik](https://github.com/user-attachments/assets/ffbb4ffd-2505-4839-832d-d3727ce1880b)

@proddy What do you think? Show only ems-devices (as is) or list all devices?

**Reply by proddy** [reply]
> @proddy What do you think? Show only ems-devices (as is) or list all devices?

Yes, if it makes integration easier.

**Comment by tp1de** [comment]
Your idea is great @MichaelDvP but I will run into the same problems as before when using different versions of the ioBroker adapter in combination with different ems-esp versions.  I am astonished that a lot of users are using stable versions of ems-esp but latest versions of the adapter or the other way around. It is difficult to keep all combinations running stable.

In the meantime I adjusted the ioBroker code for temperaturesensors to avoid the errors. But I can't test since I do not have dallas sensors. 
You can install from Github.

But on a midterm view, your proposal will make my life much easier to maintain the ioBroker adapter.

**Reply by MichaelDvP** [reply]
Confirm the new adapter from GH is working with v.3.7.0dev39 without the error message.  Not tested with old emsesp-version. (with `dallassensor`).
After first reading of the values i get for the "devices" entries only the reading of values:
```
2024-09-25 16:15:05.700 DEBUG 2168: [command] Fetched device entity/attributes for temperaturesensor/values
2024-09-25 16:15:05.822 DEBUG 2169: [command] Fetched device entity/attributes for analogsensor/values
2024-09-25 16:15:05.945 DEBUG 2170: [command] Fetched device entity/attributes for scheduler/values
2024-09-25 16:15:06.070 DEBUG 2171: [command] Fetched device entity/attributes for custom/values
```
For the ioBroker checkbox settings to read sensors the single values are fetched every time:
```
2024-09-25 16:15:06.190 DEBUG 2172: [command] Fetched device entity/attributes for temperaturesensor/values
2024-09-25 16:15:06.312 DEBUG 2173: [command] Fetched device entity/attributes for temperaturesensor/boiler_backflow
2024-09-25 16:15:06.434 DEBUG 2174: [command] Fetched device entity/attributes for analogsensor/values
2024-09-25 16:15:06.554 DEBUG 2175: [command] Fetched device entity/attributes for analogsensor/relais1
2024-09-25 16:15:06.673 DEBUG 2176: [command] Fetched device entity/attributes for analogsensor/relais2
2024-09-25 16:15:06.793 DEBUG 2177: [command] Fetched device entity/attributes for custom/values
2024-09-25 16:15:06.913 DEBUG 2178: [command] Fetched device entity/attributes for custom/test
2024-09-25 16:15:07.032 DEBUG 2179: [command] Fetched device entity/attributes for custom/ram1
```
Also different entities in ioBroker for analog (gpio from checkbox, name from devices) and temperature (i have only one attached, listed with id (from checkbox) and name(from devices)):
![grafik](https://github.com/user-attachments/assets/818219a6-2758-44c5-8245-94fffa618212)
![grafik](https://github.com/user-attachments/assets/f7a61dc4-4ea6-4136-9e4e-d0596f843fdc)

Don't think this is a problem, just want to mention that there are differences.

**Comment by tp1de** [comment]
Thanks Michael for clarifying and testing. I would highly appreciate your solution by integrating into devices.
If @proddy agrees, this should be done within a defined version of ems-esp where I could link the adapter to.
(I need then to remove the options). 

What is your timetable to create the new 3.7 stable version? Could this new entity list be included?
If yes, i would create a new ioBroker version which starts from 3.7 stable onwards and remove all codings for old versions.

**Reply by proddy** [reply]
Yes agree. I was going to release 3.7.0 last weekend but got caught up. Still to do is
 - update emsesp.org with the latest doc on 3.7.0. A lot of settings have been added and moved around
 - test with help from Kees 3.6.4 -> 3.7.0 and other variants. It's still not 100% solid
 - update the Flash tool (in progress)

hopefully this weekend we can get it out.

Also, is there anything we can add to EMS-ESP that will help? Like choose which interface version of `ioBroker.ems-esp` it should use? Or a specific endpoint where EMS-ESP can query the version of the interface?

**Reply by MichaelDvP** [reply]
I'll make a PR to dev. 
The ioBroker adapter also reads `fullname` from the value_info, i sync the common values, so every  value_info shows `name`, `fullname`, `readable`, `visible`.
